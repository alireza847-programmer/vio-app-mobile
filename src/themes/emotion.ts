import {EmotionThemeType} from 'types/themes/emotion';
import {hp, wp} from 'utils/dimensions';

const Fonts = {
  bold: 'Inter-Bold',
  medium: 'Inter-Medium',
  semiBold: 'Inter-SemiBold',
  regular: 'Inter-Regular',
};

export const theme: EmotionThemeType = {
  colors: {
    backgroundColor: '#F2F2F2',
    textColor: {
      primary: '#2A333D',
      placeHolder: '#A3ACB8',
      button: {
        primary: '#FFFFFF',
        secondary: '#0071F3',
        error: '#D83B3B',
      },
    },
    button: {
      primary: '#0071F3',
      secondary: '#F7FBFF',
    },
    stroke: {
      button: '#DAE9FA',
      divider: '#EFF2F6',
      input: '#BED2E9',
    },
  },
  fonts: {
    bold21: {
      fontFamily: Fonts.bold,
      fontSize: 21,
    },
    medium15: {
      fontFamily: Fonts.medium,
      fontSize: 15,
    },
    semiBold16: {
      fontFamily: Fonts.semiBold,
      fontSize: 16,
    },
    semiBold18: {
      fontFamily: Fonts.semiBold,
      fontSize: 18,
    },
    regular16: {
      fontFamily: Fonts.regular,
      fontSize: 16,
    },
  },
  verticalRem: hp(0.5),
  HorizontalRem: wp(1),
  radius: wp(1),
};
