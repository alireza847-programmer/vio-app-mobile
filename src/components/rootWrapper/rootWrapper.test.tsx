import React from 'react';
import {render} from '@testing-library/react-native';
import RootWrapper from './index';
import {describe, it, expect} from '@jest/globals';
import {ThemeProvider} from '@emotion/react';
import {theme} from 'themes/emotion';

describe('RootWrapper component', () => {
  it('matches snapshot', () => {
    const {toJSON} = render(
      <ThemeProvider theme={theme}>
        <RootWrapper />
      </ThemeProvider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
