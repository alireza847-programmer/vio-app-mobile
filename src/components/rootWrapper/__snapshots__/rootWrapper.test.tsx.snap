// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootWrapper component matches snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#F2F2F2",
        "flex": 1,
        "paddingHorizontal": 30,
      },
      undefined,
    ]
  }
/>
`;
